test/suite/test_compact10.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-present MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import random, time, wttest
+from helper import copy_wiredtiger_home
+from wiredtiger import stat
+from wtscenario import make_scenarios
+
+# test_compact10.py
+# Verify database can be recovered when a crash occurs during compaction.
+# It is worth noting that this test has a random behavior and one run from the other may not execute
+# the same code paths.
+class test_compact10(wttest.WiredTigerTestCase):
+    create_params = 'key_format=i,value_format=S,allocation_size=4KB,leaf_page_max=32KB'
+    conn_config = 'cache_size=100MB,statistics=(all)'
+    uri_prefix = 'table:test_compact10'
+
+    table_numkv = 100 * 1000
+    # Create up to 3 tables.
+    n_tables = random.randint(1, 3)
```

Comment:
Does this need to be random?

Reply:
My intention was to create N tables and then pick a random one when crashing. I was hoping this would bring a bit of randomness as we may compact M tables and crash at the M+1th.
This is probably not useful but I would still like to find a way to make the test random so we end up with different states when the crash occurs.

Reply:
I think 
```
        # Wait for a random table to be compacted before crashing.
        uri_idx = random.randint(0, self.n_tables - 1)
```
does what you intend for randomness.